### PR TITLE
Initial usage guide

### PR DESCRIPTION
**Pull request by kvigen**
This is the start of a usage guide. A lot stuff still needs to be filled out, but this is at least a start.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,22 +1,57 @@
...
 sWAGger - Web API Generator
 
-# Running
+## Usage
+### Generating Code
+Create a swagger.yml file with your [service definition](http://editor.swagger.io/#/). Note that WAG supports a [subset](https://github.com/Clever/wag#swagger-spec) of the Swagger spec. Then generate your code:
 ```
-go run main.go genclients.go
-cd impl/ && ./impl
+make swagger
```
Comment (rgarcia) [inline README.md]: @kvigen mind if I edit this directly to show how to set up your makefile targets and utilize swagger.mk?

Comment (kvigen) [inline README.md]: go for it

File: README.md [hunk 2]
```diff
@@ -1,22 +1,141 @@
...
+Then generate your code:
 ```
-make test
+make generate
+```
+
+This generates four directories. You should not have to modify any of the generated code:
+- generated/models: contains all the definitions in your Swagger file as well as API input / output definitions
+- generated/server: contains the router, middleware, and handler logic
+- generated/clients: contains the Go client library
```
Comment (rgarcia) [inline README.md]: I think this is `generated/client`?

**Comment by rgarcia** [comment]
@kvigen updated this with node instructions, also had a question about a directory name. Otherwise lgtm :shipit:

File: README.md (hunk 2 above)
Comment (kvigen) [inline README.md]: derp, good catch

**Comment by kvigen** [comment]
thanks, the JS stuff looks good

**Pull request merged**